### PR TITLE
Backport d23ef9e to release/18.x

### PR DESCRIPTION
**Pull request by Sirraide**
Manually cherry-pick and backport https://github.com/llvm/llvm-project/commit/d23ef9ef3685eb42ebf719bc28cfe2e4651932fc as discussed in https://github.com/llvm/llvm-project/pull/83103.

CC @AaronBallman, @tstellar

### CODE REVIEW

**Review by AaronBallman — APPROVED** [review]
LGTM assuming the release managers are also happy with it.

**Comment by tstellar** [comment]
Merged: d8352e93c1c8042d9166eab3d76d6c07ef585b6d

**Pull request closed without merging**
